inference-engine/src/mkldnn_plugin/cpu_blocked_memory_desc.cpp

```diff
@@ -0,0 +1,246 @@
+        IE_THROW() << "BlockedMemoryDesc do not support undefined blocks.";
+    }
+
+    this->order = order;
+    this->blockedDims = blockedDims;
+    this->offsetPadding = offsetPadding;
+
+    if (offsetPaddingToData.empty() && !order.empty()) {
+        this->offsetPaddingToData.resize(order.size());
+        this->offsetPaddingToData[order.size() - 1] = 0;
+        for (size_t i = 2; i <= order.size(); i++) {
+            this->offsetPaddingToData[order.size() - i] = 0;
+        }
+    } else {
+        this->offsetPaddingToData = offsetPaddingToData;
+    }
+
+    if (strides.empty() && !order.empty()) {
+        if (std::any_of(this->blockedDims.begin(), this->blockedDims.end(), [](size_t val) { return val == Shape::UNDEFINED_DIM; })) {
+            this->strides.resize(order.size(), Shape::UNDEFINED_DIM);
+        } else {
+            this->strides.resize(order.size());
+            this->strides[order.size() - 1] = 1;
+            for (size_t i = 2; i <= order.size(); i++) {
+                this->strides[order.size() - i] = this->strides[order.size() - (i - 1)] * this->blockedDims[blockedDims.size() - (i - 1)];
+            }
+        }
+    } else {
+        this->strides = strides;
+    }
+}
+
+bool BlockedMemoryDesc::isDefined() const {
+    bool defined = true;
+    defined = defined && std::none_of(blockedDims.cbegin(), blockedDims.cend(), [](size_t val) { return val == Shape::UNDEFINED_DIM; });
+    defined = defined && std::none_of(strides.cbegin(), strides.cend(), [](size_t val) { return val == Shape::UNDEFINED_DIM; });
+    defined = defined && std::none_of(order.cbegin(), order.cend(), [](size_t val) { return val == Shape::UNDEFINED_DIM; });
+    defined = defined && std::none_of(offsetPaddingToData.cbegin(), offsetPaddingToData.cend(), [](size_t val) { return val == Shape::UNDEFINED_DIM; });
+    defined = defined && offsetPadding != Shape::UNDEFINED_DIM;
+
+    return defined;
+}
+
+bool BlockedMemoryDesc::isCompatible(const MemoryDesc& rhs) const {
+    const MemoryDesc* pRhs = &rhs;
+    if (auto blockingDesc = dynamic_cast<const BlockedMemoryDesc*>(pRhs)) {
+        return isCompatible(*blockingDesc);
+    } else if (auto mkldnnDesc = dynamic_cast<const MKLDNNMemoryDesc*>(pRhs)) {
+        return mkldnnDesc->isCompatible(*this);
+    } else {
+        return false;
+    }
+}
+
+bool BlockedMemoryDesc::isCompatible(const BlockedMemoryDesc& rhs) const {
+    if (this->getShape() != rhs.getShape() || this->getPrecision() != rhs.getPrecision())
+        return false;
+
+    if (!isEqualOrUndefined(this->getBlockDims(), rhs.getBlockDims())) {
+        return false;
+    }
+
+    if (!isEqualOrUndefined(this->getOffsetPaddingToData(), rhs.getOffsetPaddingToData())) {
+        return false;
+    }
+
+    size_t skipAxis = this->getShape().getRank() > 0 && this->getShape().getDims().front() == 1 ? 0 :
```

Comment:
This is a legacy behavior, which has performance reasons. Previously it was placed in the predecessor of this function called `initTensorsAreEqual`. This WA allows to avoid unnecessary reorders if the memory is used in place and the batch size is equal to 1.

Reply:
We need to add a comment about why we use this approach.